.github/workflows/versioned-images.yml

```diff
@@ -0,0 +1,87 @@
+name: Build and Publish Versioned Images
+
+on:
+  workflow_dispatch:
```

Comment:
Can we perhaps at least start with taking input the implementation name maybe? So that we're not building all versions for *all* implementations, we just select from the list of implementations containing version files as you have it? Or does that complicate things a lot?